lib/routes/account.js

```diff
@@ -1257,49 +1257,116 @@ module.exports = function (
+                  {
+                    authSalt: authSalt,
+                    verifyHash: verifyHash,
+                    wrapWrapKb: crypto.randomBytes(32),
+                    verifierVersion: password.version
+                  }
+                )
+              }
+            )
+            .then(
+              function () {
+                // Notify all devices that the account has changed.
+                push.notifyUpdate(accountResetToken.uid, 'passwordReset')
+
+                return db.account(accountResetToken.uid)
+              }
+            )
+            .then(
+              function (accountData) {
+                account = accountData
+                log.activityEvent('account.reset', request, {
+                  uid: account.uid.toString('hex')
+                })
+                log.event(
+                  'reset',
+                  {
+                    uid: account.uid.toString('hex') + '@' + config.domain,
+                    generation: account.verifierSetAt
+                  }
+                )
+                return customs.reset(account.email)
+              }
+            )
+            .then(
+              function () {
+                return password.unwrap(account.wrapWrapKb)
+              }
+            )
+            .then(
+              function (wrapKbData) {
+                wrapKb = wrapKbData
+              }
+            )
+        }
+
+        function createSessionToken () {
+          // Create a verified sessionToken so that the client does
+          // not have to re-login
+          var sessionTokenOptions = {
+            uid: account.uid,
+            email: account.email,
+            emailCode: account.emailCode,
+            emailVerified: account.emailVerified,
+            verifierSetAt: account.verifierSetAt
+          }
+
+          return db.createSessionToken(sessionTokenOptions, request.headers['user-agent'])
+            .then(
+              function (result) {
+                sessionToken = result
+              }
+            )
+        }
+
+        function createKeyFetchToken () {
+          if (wantsKeys(request)) {
+            return db.createKeyFetchToken({
+                uid: account.uid,
+                kA: account.kA,
+                wrapKb: wrapKb,
+                emailVerified: account.emailVerified
               })
-              log.event(
-                'reset',
-                {
-                  uid: account.uid.toString('hex') + '@' + config.domain,
-                  generation: account.verifierSetAt
+              .then(
+                function (result) {
+                  keyFetchToken = result
                 }
               )
-              return customs.reset(account.email)
-            }
-          )
-          .then(
-            function () {
-              return {}
-            }
-          )
-          .done(reply, reply)
+          }
+        }
+
+        function createResponse () {
+          var response = {
```

Comment:
If sessionToken creation has not been opted in to, I propose the response should be `{}`, like it used to be.